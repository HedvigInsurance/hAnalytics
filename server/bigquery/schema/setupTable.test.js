const setupTable = require("./setupTable");
const createBigQueryConfigMock = require("../config.mock");

test("sets up a table correctly", async () => {
  const bigQueryConfig = createBigQueryConfigMock();

  await setupTable(
    "mock_table",
    "description",
    [
      {
        name: "property",
        type: "STRING",
      },
    ],
    bigQueryConfig
  );

  expect(bigQueryConfig.bigquery.getTables()).toMatchSnapshot();
});

test("updates a table correctly", async () => {
  const bigQueryConfig = createBigQueryConfigMock();

  await setupTable(
    "mock_table",
    "description",
    [
      {
        name: "property",
        type: "STRING",
      },
    ],
    bigQueryConfig
  );

  expect(bigQueryConfig.bigquery.getTables().length).toEqual(1);

  await setupTable(
    "mock_table",
    "description",
    [
      {
        name: "property",
        type: "STRING",
      },
      {
        name: "property_2",
        type: "STRING",
      },
    ],
    bigQueryConfig
  );

  expect(bigQueryConfig.bigquery.getTables()).toMatchSnapshot();
});
