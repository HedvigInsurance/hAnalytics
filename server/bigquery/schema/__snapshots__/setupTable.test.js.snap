// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets up a table correctly 1`] = `
Array [
  Object {
    "config": Object {
      "description": "description",
      "schema": Object {
        "fields": Array [
          Object {
            "name": "property",
            "type": "STRING",
          },
        ],
      },
      "timePartitioning": Object {
        "expirationMS": "7776000000",
        "type": "DAY",
      },
    },
    "name": "mock_table",
    "rows": Array [],
  },
]
`;

exports[`updates a table correctly 1`] = `
Array [
  Object {
    "config": Object {
      "description": "description",
      "schema": Object {
        "fields": Array [
          Object {
            "name": "property",
            "type": "STRING",
          },
          Object {
            "name": "property_2",
            "type": "STRING",
          },
        ],
      },
      "timePartitioning": Object {
        "expirationMS": "7776000000",
        "type": "DAY",
      },
    },
    "name": "mock_table",
    "rows": Array [],
  },
]
`;
